proxy/logging/LogUtils.cc

```diff
@@ -571,3 +587,158 @@ LogUtils::file_is_writeable(const char *full_filename, off_t *size_bytes, bool *
 
   return ret_val;
 }
+
+namespace
+{
+// Get a string out of a MIMEField using one of its member funcitions, and put it into a buffer writer, terminated with a nul.
+//
+void
+marshalStr(ts::FixedBufferWriter &bw, const MIMEField &mf, const char *(MIMEField::*get_func)(int *length) const)
+{
+  int length;
+  const char *data = (mf.*get_func)(&length);
+
+  if (!data or (*data == '\0')) {
+    // Empty string.  This is a problem, since it would result in two successive nul characters, which indicates the end of the
+    // marshaled hearer.  Change the string to a single blank character.
+
+    static const char Blank[] = " ";
+    data                      = Blank;
+    length                    = 1;
+  }
+
+  bw << std::string_view(data, length) << '\0';
+}
+
+void
+unmarshalStr(ts::FixedBufferWriter &bw, const char *&data)
+{
+  bw << '{';
+
+  while (*data) {
+    bw << *(data++);
+  }
+
+  // Skip over terminal nul.
+  ++data;
+
+  bw << '}';
+}
+
+} // end anonymous namespace
+
+namespace LogUtils
+{
+// Marshals header tags and values together, with a single terminating nul character.  Returns buffer space required.  'buf' points
+// to where to put the marshaled data.  If 'buf' is null, no data is marshaled, but the function returns the amount of space that
+// would have been used.
+//
+int
+marshalMimeHdr(MIMEHdr *hdr, char *buf)
+{
+  std::size_t bwSize = buf ? SIZE_MAX : 0;
+
+  ts::FixedBufferWriter bw(buf, bwSize);
```

Comment:
So, the marshaling functions gets called twice, once with a buf == nullptr;. What happens with this for this case? It becomes a malloc(SIZE_MAX) ?

Reply:
No, BufferWriter never allocates memory. It will have size 0 and dump all of the output, while tracking the amount of output sent (which is retrieved by `extent`).

Reply:
To answer myself, yes it does, but looking at the other logging code, I see it all does that, so this is effectively doing the same thing as all the existing code.